### ISSUE TITLE
 I would like to save firefox web page shortcuts to home screen. Safari has a Save to home screen option, but I can not find the same option in Fire Fox.

### ISSUE THREAD

**Issue by SimonBasca** [mozilla-mobile/firefox-ios#5347]
Logged via: https://bugzilla.mozilla.org/show_bug.cgi?id=1572707

Steps to reproduce:

I would like to save firefox web page shortcuts to home screen. Safari has a Save to home screen option when you tap the share icon, but I can not find the same option in Fire Fox.

Actual results:

Firefox has no option to save web page shortcut to desktop

Expected results:

When user taps the ellipsis in the upper right of the screen and then tap the Share Page With icon, they should have a Save to Home Screen option as in safari.

I have included a safari example pic showing the option that does not appear in firefox

**Comment by AH-SWG** [comment]
This issue is not fixed and is still prevalent in iOS, Firefox version 105.0 (18490)

**Comment by aleksandersumowski** [comment]
Was it actually implemented?

**Comment by AH-SWG** [comment]
@aleksandersumowski No, it never was. I'm guessing @dnarcese thought it was taking up too much screen space in the open queue?

**Comment by lmarceau** [comment]
In case this helps; There's a way to save to the homepage through the menu with "Add to Shortcuts". In this example I added to shortcut the apple store website to my shortcuts. It then appears on my homepage.
![Screen Shot 2023-01-31 at 9 37 47 AM](https://user-images.githubusercontent.com/11338480/215790206-e7a0a3af-6148-4812-a4dc-11c2c24d435f.png)
![Screen Shot 2023-01-31 at 9 43 39 AM](https://user-images.githubusercontent.com/11338480/215791715-db4105e1-15f3-43ca-9ae9-dca33aaec7e1.png)

Not like Safari as it's not through the Share menu but rather our normal menu. Result in both case is that the website is saved on the homepage.

**Comment by AH-SWG** [comment]
> Result in both case is that the website is saved on the homepage.

No, the result in Safari is that the shortcut is saved to the iOS desktop like a standalone app is. It's not even close to the same feature. Saying the result is the same is like saying that bookmarking a web page is the same as adding it to the Desktop of a computer.

**Comment by OrlaM** [comment]
@lmarceau The feature request refers to desktop which I think means the iPhone home screen and not the Firefox home page. On Safari using the share sheet you can save a site to the iPhone home screen so that it appears as if it is an app. It's a legacy feature from way back when there was no app store and all "apps" were just websites. I'm not even sure we have the ability to replicate this behaviour or if it's a Safari privileged thing, I'd need to do some digging to verify.

**Comment by lmarceau** [comment]
Oh I see, I had tried it and saw it in my Safari homepage and thought that was it, sorry about that. I don't think this feature is possible to implement by third party app developers like us. Unless I missed it (which I would be more than happy to have missed), I don't think those APIs are made available by Apple. The best we can do is through widgets.

**Comment by guicbrito** [comment]
Looks like it was’t a legacy feature.
See https://webkit.org/blog/13878/web-push-for-web-apps-on-ios-and-ipados/
I’m +1 for the feature. Would be cool to pin ChatGPT to my home screen.

**Comment by lordzsolt** [comment]
Would be nice to know why this issue was closed without being implemented...

**Comment by DumbledoreCalrissian** [comment]
Legacy? So what?
This is a feature almost all browsers on all operating systems have!

**Comment by 4cq2** [comment]
https://github.com/mozilla-mobile/firefox-ios/issues/14028

**Comment by OrlaM** [comment]
There is a video in this PR showing how to use this feature https://github.com/mozilla-mobile/firefox-ios/pull/14174
If you do not see the option used in the video then please open a bug ticket, it would be helpful to include screenshots or videos of what you are seeing instead, also what version of the app and OS you are using.

**Comment by DRA6N** [comment]
Yeah it definitely isn't there. I'll open a bug ticket.

**Comment by lmarceau** [comment]
Having empty squares means that there are no tiles to show for that space. So for example, if you rotate your screen to landscape, you'll have even more blank spaces as there are more tiles showing in landscape than portrait.

**Comment by gerroon** [comment]
Thanks but how does one add shortcuts/urls to those tiles? I use Brave as well on IOS, and it has the same type of grid widget, its widget is full of urls that I favorited, so it is conpletefly functional.

> Having empty squares means that there are no tiles to show for that space. So for example, if you rotate your screen to landscape, you'll have even more blank spaces as there are more tiles showing in landscape than portrait.

**Comment by lmarceau** [comment]
You can add shortcuts by navigating to the website you want to add, then from the menu select the "Add to shortcuts" option. Let me know if there's anything!

**Comment by gerroon** [comment]
I can add shortcuts and I can use the shortcuts but they do not show up in the grid. 

See the image, it is all empty boxes.

![image](https://github.com/mozilla-mobile/firefox-ios/assets/8519469/ac25d62b-b0a1-4148-aa79-f68b43ac612d)


> You can add shortcuts by navigating to the website you want to add, then from the menu select the "Add to shortcuts" option. Let me know if there's anything!
